client/state/analytics/test/middleware.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+
+import useFakeDom from 'test/helpers/use-fake-dom';
+import useMockery from 'test/helpers/use-mockery';
+
+import {
+	withAnalytics,
+	bumpStat,
+	recordGoogleEvent,
+	recordGooglePageView,
+	recordTracksEvent,
+	recordPageView
+} from '../actions';
+import analyticsMock from './analytics-mock';
```

Comment:
@dmsnell thanks for moving files :)

We are about to drop whitelist config `tests.json` for `server` and `test` top level folders with #4673. `client` folder needs to wait until we migrate all test folders using `Makefile`. Test discovery is going to be based in the end on `glob` pattern similar to: `@(client|server|test)/**/test/*.@(js|jsx)`. That's why we think it's better to keep all non-test files in the `test` subfolder directly. We should document it somewhere once we drop `test.json` files completely ;)

I'm busy with support rotation this week, but this PR is really great idea :)